owncloudApp/src/main/java/com/owncloud/android/presentation/sharing/shares/PublicShareDialogFragment.kt

```diff
@@ -310,6 +320,18 @@ class PublicShareDialogFragment : DialogFragment() {
         }
     }
 
+    private fun initPasswordChangeInputListener() {
+        binding.shareViaLinkPasswordValue.doOnTextChanged { text, _, _, _ ->
+            capabilities?.passwordPolicy?.let { passwordPolicy ->
+                requirementsPasswordPolicy(text.toString(), passwordPolicy)
+            } ?: binding.saveButton.also {
```

Comment:
There are 2 things that don't make sense here:
1. The use of `also`. Why are you using that Kotlin expression when you need `saveButton` just once?
2. The Elvis (`?:`) operator. Why just executing the second part when `passwordPolicy` is null?

Reply:
Here we use the elvis operator (`?:`) to guarantee we run one of the conditional branches. If property exists, then we can capture and use its value, if the value is null we can ensure we do another logic. In this way, we collect the logic that will be executed when it is an oc10 server.  I have created a new function to handle `null` `passwordPolicy`. In this case, we do not need to use `run` function to get a block of code.